### PR TITLE
systemd-service: build rust-agent systemd services

### PR DESCRIPTION
**Pull request by Pennyzct**
I add another sub-command `build-service` in Makefile to generate rust-agent-related systemd service files, which are necessary for building guest rootfs image.
The whole design is following the one in go-agent. 
And since I just copy the file [kata-conatiners.target](https://github.com/kata-containers/agent/blob/master/kata-containers.target) and [kata-agent.service](https://github.com/kata-containers/agent/blob/master/kata-agent.service.in) and paste them here, I don't change the copyright info in the header. Hope I'm doing the right thing???😜

### CODE REVIEW

**Comment by Pennyzct** [comment]
### Updates:
Travis-ci failed on the following error:
```
error[E0133]: use of mutable static is unsafe and requires unsafe function or block
  --> /root/.cargo/git/checkouts/path-absolutize-0aead0458025157b/8893638/src/unix.rs:15:35
   |
15 |             let path = Path::join(&CWD, self);
   |                                   ^^^^ use of mutable static
   |
   = note: mutable statics can be mutated by multiple threads: aliasing violations or data races will cause undefined behavior
error: aborting due to previous error
```
And I've raised another commit here to fix this error. ;)

**Comment by yyyeerbo** [comment]
we probably need to fix the version of path-absolute in rust-agent..

**Comment by Pennyzct** [comment]
Hi~ @yyyeerbo 
I've already added one commit here to fix the `path-absolutize` error. Plz take a review~~~~

**Comment by Pennyzct** [comment]
Hi~~ @yyyeerbo 
The import of `path-absolutize` may only locate in crate `rustjail` and I‘ve already raised an commit here to do the version update.
And travis ci is happy now~~~~😊

File: src/agent/kata-agent.service.in [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2018-2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+[Unit]
+Description=Kata Containers Agent
+Documentation=https://github.com/kata-containers/agent
```
Comment (jodh-intel) [inline src/agent/kata-agent.service.in]: Nit: Please could you change this URL to https://github.com/kata-containers/kata-containers

**Comment by Pennyzct** [comment]
Hi guys~ @yyyeerbo @lifupan 
Ubuntu CI failed constantly. The error is as follows: 
```
install: cannot stat '/tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/kata-containers/src/agent/target/x86_64-unknown-linux-musl/debug/kata-agent': No such file or directory
Failed at 23: "${cidir}/install_kata_image_rust.sh"
```
Looks to me, the outputs show that rust agent was not built successfully.
Digging in test scripts and it should be built [here](https://github.com/kata-containers/tests/blob/master/.ci/install_kata_image_rust.sh#L24):
```
	pushd "${GOPATH}/src/${rust_agent_repo}/src/agent"
	rustup target add "${arch}"-unknown-linux-musl
	echo "Building rust agent"
	make -j $(nproc)
	popd
```
But seeing from the outputs, the building part shows no compiling info, which is odd.
```
/tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/kata-containers/src/agent /tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests
info: component 'rust-std' for target 'x86_64-unknown-linux-musl' is up to date
Building rust agent
/tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests

```

**Comment by lifupan** [comment]
> Hi guys~ @yyyeerbo @lifupan
> Ubuntu CI failed constantly. The error is as follows:
> 
> ```
> install: cannot stat '/tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/kata-containers/src/agent/target/x86_64-unknown-linux-musl/debug/kata-agent': No such file or directory
> Failed at 23: "${cidir}/install_kata_image_rust.sh"
> ```
> 
> Looks to me, the outputs show that rust agent was not built successfully.
> Digging in test scripts and it should be built [here](https://github.com/kata-containers/tests/blob/master/.ci/install_kata_image_rust.sh#L24):
> 
> ```
> 	pushd "${GOPATH}/src/${rust_agent_repo}/src/agent"
> 	rustup target add "${arch}"-unknown-linux-musl
> 	echo "Building rust agent"
> 	make -j $(nproc)
> 	popd
> ```
> 
> But seeing from the outputs, the building part shows no compiling info, which is odd.
> 
> ```
> /tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/kata-containers/src/agent /tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests
> info: component 'rust-std' for target 'x86_64-unknown-linux-musl' is up to date
> Building rust agent
> /tmp/jenkins/workspace/kata-containers-rust-agent-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests
> ```

Hi @Pennyzct 

Since that ci script build kata-agent without specify the specific "default" target, and 
when no target is specified on the command line make builds the default goal.

The default goal is the first target , but you added a new target which on top of the "default" target, that's why the CI didn't build the kata-agent. 

You can move the "default" target on the top  of the target you added and it would  work.

File: src/agent/kata-agent.service.in (hunk 1 above)
Comment (Pennyzct) [reply src/agent/kata-agent.service.in]: thanks! good catch🤑. updated~

**Comment by Pennyzct** [comment]
Hi~ @lifupan 
Thanks thousand times for the debugging and explanation. ;) 🐹 already updated~~

**Pull request merged**